src/resources/app/components/AnnouncementContent.vue

```diff
@@ -0,0 +1,159 @@
+<template>
+  <div class="px-6 py-4">
+    <div class="d-flex justify-between">
+      <div class="d-flex">
+        <v-avatar class="bg-primary mr-4" id="avatar" size="48"></v-avatar>
+        <div>
+          <div id="author">
+            {{ name }}
+          </div>
+          <div
+            id="date"
+            class="text-sm text-textGray">
+            {{ announcementDate }}
+          </div>
+        </div>
+      </div>
+      <v-menu
+        id="menu"
+        bottom
+        right
+      >
+        <template v-slot:activator="{ on, attrs }">
+          <v-btn
+            icon
+            v-bind="attrs"
+            v-on="on"
+          >
+            <v-icon>mdi-dots-vertical</v-icon>
+          </v-btn>
+        </template>
+
+        <v-list>
+          <v-list-item
+            link
+            v-for="(item, i) in items"
+            :key="i"
+          >
+            <v-list-item-title
+              :class="item.class"
+              @click="isEditing=true"
+            >{{ item.title }}
+            </v-list-item-title>
+          </v-list-item>
+        </v-list>
+      </v-menu>
+    </div>
+    <div
+      id="content"
+      class="mt-4"
+      v-if="!isEditing"
+    >
+      {{ content }}
+    </div>
+    <div
+      id="edit-form"
+      v-if="isEditing"
+    >
+      <v-form
+        v-model="isFormValid"
+        ref="form"
+        class="mx-6 my-4"
+      >
+        <div>
+          <v-textarea
+            outlined
+            v-model="announcement"
+            :rules="announceRules.concat(lengthRules)"
+            :counter="65535"
+            label="Announcement"
+            placeholder="Announcement for everyone..."
+          ></v-textarea>
+        </div>
+        <div class="text-right">
+          <v-btn
+            depressed
+            color=light
+            @click="isEditing = false; announcement = content"
+            class="mr-4"
+          >
+            <div
+              id="cancel-button"
+              class="text-primary"
+            >
+              Cancel
+            </div>
+          </v-btn>
+          <v-btn
+            id="post-button"
+            depressed
+            color=primary
+            @click="editAnnouncement"
+            :disabled="!isFormValid"
+          >
+            Post
+          </v-btn>
+        </div>
+      </v-form>
+    </div>
+  </div>
+</template>
+
+<script>
+import {EditAnnouncement} from "../../api/announcement/EditAnnouncement";
+
+export default {
+  name: "AnnouncementContent",
+  props: [
+    "id",
+    "name",
+    "date",
+    "content",
+  ],
+  data() {
+    return {
+      announcementDate: this.dateSlicing(),
```

Comment:
This can be placed in `computed` instead
ref: https://v2.vuejs.org/v2/guide/computed.html